src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -326,15 +328,8 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     internal val modifiers: Set<KModifier> = emptySet()
   ) {
 
-    val isEnum get() = this is Class && ENUM in modifiers
-
-    val isAnnotation get() = this is Class && ANNOTATION in modifiers
-
-    internal val isSimpleClass get() = this is Class && !isEnum && !isAnnotation
-
-    internal val implicitPropertyModifiers get() =
-      if (isAnnotation) emptySet()
-      else defaultImplicitPropertyModifiers + when {
+    internal val implicitPropertyModifiers get() = defaultImplicitPropertyModifiers + when {
+        ANNOTATION in modifiers -> emptySet()
```

Comment:
I feel like this isn't correct, but it's unrelated to your PR. An expect annotation should get the implicits.

Reply:
Guess you're right. I'll try to write a failing test and will file an issue.